File: CF450A.py
import math

n, m = input().split(" ")
hunger = input().split(" ")

max = 0
pos = 0
for i, hungery in enumerate(hunger):
    weight = math.ceil(int(hungery) / int(m) ) * int(n) + int(i)
    if(weight >= max):
        max = weight
        pos = i

print(pos + 1)

File: CF450A.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n, m, i, pos = 0, weight;
    int hunger[100], max_hunger_weight = -1;
    
    cin >> n >> m;
    for (i = 0; i < n; i++) {
        cin >> hunger[i];
        weight = (ceil((float)hunger[i] / m)) * n + i;
        
        if (weight >= max_hunger_weight) {
            max_hunger_weight = weight;
            pos = i;
        }
    }

    cout << pos + 1 << endl;
    return 0;
}
